pygrametl/__init__.py

```diff
@@ -444,23 +444,38 @@ def now(ignoredtargetconn=None, ignoredrow=None, ignorednamemapping=None):
 
 
 def ymdparser(ymdstr):
-    """Convert a string of the form 'yyyy-MM-dd' to a datetime.date.
+    """Convert input of the form 'yyyy-MM-dd' to a datetime.date.
 
        If the input is None, the return value is also None.
+       If the input is a datetime.date, it is returned.
+       If the input is a datetime.datetime, its date is returned.
+       Else the input is cast to str and quotes are stripped before it is
+       split into the different parts needed to create a datetime.date.
     """
     if ymdstr is None:
         return None
+    elif isinstance(ymdstr, date):
+        return ymdstr
+    elif isinstance(ymdstr, datetime):
+        return ymdstr.date()
+    ymdstr = str(ymdstr).strip("'\"")
     (year, month, day) = ymdstr.split('-')
     return date(int(year), int(month), int(day))
 
 
 def ymdhmsparser(ymdhmsstr):
```

Comment:
Maybe remove `str` from `ymdhmsstr` as the input need not be a string, e.g., `ymdhmsornone`.

Reply:
Agreed, the name is misleading now, but changing it would break backwards compatibility if anyone has used a named parameter

Reply:
Good point, let us leave it.